specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,222 @@
+    "paymentGroup": [
+      {
+        "signature": "VxpW6emWwct7ymzyXa/F2T7rwepM71XwsslPKEdpVqmbcRUB28QMdYJuuLkeP4dnVoI7emmxT6Q/YktqHc8xBg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "fee": "2000",
+          "fv": "6631154",
+          "lv": "6632154",
+          "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "amnt": "0",
+          "type": "pay"
+        }
+      },
+      {
+        "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+          "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+          "type": "axfer",
+          "fee": "0",
+          "fv": "6631154",
+          "lv": "6632154",
+          "xaid": "168103"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification
+
+Steps to verify a payment for the `exact` scheme on Algorand:
+
+1. Verify the signature is valid for each transaction in the `paymentGroup`.
+  a. Encode the `txn` parameters as described in the encoding section.
+  b. Recover the public key of the sender from the `snd` or `asnd` field by decoding from the address.
+  c. Verify the signature against the encoded message using the recovered public key.
+2. Verify that the atomic group of transactions is valid and can be executed as a single unit.
+3. Execute the payment group against an Algorand node's `simulate` endpoint to ensure the transactions would result in the amount specified in `paymentRequirements.maxAmountRequired` being transferred to the resource server's address.
+
+
+## Settlement
+
+Once the group is validated by the resource server, settlement can occur by the facilitator submitting the `paymentGroup` to the Algorand network through the `v2/transactions` endpoint against any valid Algorand node.
+
+In Algorand there are no consensus forks and so it achieves finality instantly a transaction is included in a block. So as soon as the transaction is included in a block, the payment is considered settled and the facilitator can inform the resource server of the successful payment and proceed with the resource delivery.
+
+## Additional Considerations
+
+- In order for the resource server to receive payment on a particular Asset, it must be opted-in to that asset. In Algorand, accounts must explicitly signal they are okay with receiving a particular asset, otherwise an asset transfer to that account will be rejected. So as part of the resource server's setup, it should ensure it has opted-in to the asset ID specified in the `paymentRequirements.asset`.
```

Comment:
```suggestion
- In order for the resource server to receive payment on a particular Asset, it **MUST** be opted-in to that asset.

In Algorand, accounts explicitly enable receiving a particular asset, otherwise an asset transfer to that account will be rejected. So as part of the resource server's setup, it **SHOULD** ensure it has opted-in to the asset ID specified in the `paymentRequirements.asset`.
```